hbase-assembly/src/main/assembly/client.xml

```diff
@@ -104,24 +56,44 @@
 
   <dependencySets>
     <dependencySet>
-      <outputDirectory>lib/shaded-clients</outputDirectory>
-      <includes>
-        <include>org.apache.hbase:hbase-shaded-client</include>
-        <include>org.apache.hbase:hbase-shaded-mapreduce</include>
-        <include>org.apache.hbase:hbase-shaded-client-byo-hadoop</include>
-      </includes>
+      <outputDirectory>lib</outputDirectory>
+      <useTransitiveDependencies>true</useTransitiveDependencies>
+      <excludes>
+        <!-- Exclude the shaded jars that go in the lib/shaded-clients directory -->
+        <exclude>org.apache.hbase:hbase-shaded-client</exclude>
+        <exclude>org.apache.hbase:hbase-shaded-mapreduce</exclude>
+        <exclude>org.apache.hbase:hbase-shaded-client-byo-hadoop</exclude>
+        <!-- Exclude the Ruby jar that goes in the lib/ruby directory -->
+        <exclude>org.jruby:jruby-complete</exclude>
+        <!-- Exclude jars that go into the lib/client-facing-thirdparty directoy -->
+        <exclude>com.github.stephenc.findbugs:findbugs-annotations</exclude>
```

Comment:
so the exclude block should be kept in sync with any dependencies we add in a sub folder of lib. may be add a note somewhere to avoid dependencies getting duplicated?

Reply:
Good idea.